lib/elixir/lib/module.ex

```diff
@@ -406,8 +407,19 @@ defmodule Module do
 
       Foo.sum(1, 2) #=> 3
 
+  You can also pass a `Macro.Env` instance as the first argument,
+  which will have both the module and all options extracted from
+  the environment:
+
+      defmodule Foo do
+        contents = quote do: (def sum(a, b), do: a + b)
+        Module.eval_quoted __ENV__, contents
+      end
+
+      Foo.sum(1, 2) #=> 3
+
```

Comment:
We could merge the two sections so as to not repeat most of the `Macro.Env`-related stuff. Wdyt?